server/src/transport/ws.rs

```diff
@@ -415,6 +416,15 @@ where
 
 	match server.receive_request(&req) {
 		Ok(response) => {
+			let upgraded = match hyper::upgrade::on(req).await {
+				Ok(u) => u,
+				Err(e) => {
+					tracing::debug!(target: LOG_TARGET, "WS upgrade handshake failed: {}", e);
+					return Err(HttpResponse::new(HttpResponseBody::from(format!("WS upgrade handshake failed {e}"))));
+				}
+			};
```

Comment:
Like, previously it would only be executed when the returned future is polled, but now it's executed in the `connect(..).await` and has to finish before the fut is returned

Reply:
https://github.com/hyperium/hyper/blob/master/examples/upgrades.rs#L56-#L72

Reply:
Thus, it will be stuck until the response is returned but I can add a comment in the code because it confuses me as well